pastel doesn't build: error[E0308]: mismatched types

**Describe the bug**

Doesn't build.

```
error[E0308]: mismatched types
  --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/bhcomp.rs:62:24
   |
62 |     let bytes = bits / Limb::BITS;
   |                        ^^^^^^^^^^ expected `usize`, found `u32`

error[E0277]: cannot divide `usize` by `u32`
  --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/bhcomp.rs:62:22
   |
62 |     let bytes = bits / Limb::BITS;
   |                      ^ no implementation for `usize / u32`
   |
   = help: the trait `Div<u32>` is not implemented for `usize`

error[E0308]: mismatched types
   --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/bigcomp.rs:158:55
    |
158 |     let nlz = den.leading_zeros().wrapping_sub(wlz) & (u32::BITS - 1);
    |                                                       ^^^^^^^^^^^^^^^ expected `usize`, found `u32`

error[E0277]: no implementation for `usize & u32`
   --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/bigcomp.rs:158:53
    |
158 |     let nlz = den.leading_zeros().wrapping_sub(wlz) & (u32::BITS - 1);
    |                                                     ^ no implementation for `usize & u32`
    |
    = help: the trait `BitAnd<u32>` is not implemented for `usize`

error[E0308]: mismatched types
   --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/bigcomp.rs:176:40
    |
176 |         let (q, r) = shift.ceil_divmod(Limb::BITS);
    |                                        ^^^^^^^^^^ expected `usize`, found `u32`
    |
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
    |
176 |         let (q, r) = shift.ceil_divmod(Limb::BITS.try_into().unwrap());
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1049:42
     |
1049 |     let mut count = index.saturating_mul(Limb::BITS);
     |                                          ^^^^^^^^^^ expected `usize`, found `u32`
     |
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
     |
1049 |     let mut count = index.saturating_mul(Limb::BITS.try_into().unwrap());
     |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1064:28
     |
1064 |     Limb::BITS.checked_mul(x.len())
     |                            ^^^^^^^ expected `u32`, found `usize`
     |
help: you can convert a `usize` to a `u32` and panic if the converted value doesn't fit
     |
1064 |     Limb::BITS.checked_mul(x.len().try_into().unwrap())
     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1065:22
     |
1065 |         .map(|v| v - nlz)
     |                      ^^^ expected `u32`, found `usize`

error[E0277]: cannot subtract `usize` from `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1065:20
     |
1065 |         .map(|v| v - nlz)
     |                    ^ no implementation for `u32 - usize`
     |
     = help: the trait `Sub<usize>` is not implemented for `u32`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1066:20
     |
1066 |         .unwrap_or(usize::max_value())
     |                    ^^^^^^^^^^^^^^^^^^ expected `u32`, found `usize`
     |
help: you can convert a `usize` to a `u32` and panic if the converted value doesn't fit
     |
1066 |         .unwrap_or(usize::max_value().try_into().unwrap())
     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1064:5
     |
1060 |   pub fn bit_length(x: &[Limb]) -> usize {
     |                                    ----- expected `usize` because of return type
...
1064 | /     Limb::BITS.checked_mul(x.len())
1065 | |         .map(|v| v - nlz)
1066 | |         .unwrap_or(usize::max_value())
     | |______________________________________^ expected `usize`, found `u32`
     |
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
     |
1064 |     Limb::BITS.checked_mul(x.len())
1065 |         .map(|v| v - nlz)
1066 |         .unwrap_or(usize::max_value()).try_into().unwrap()
     |

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1091:23
     |
1091 |     debug_assert!(n < bits && n != 0);
     |                       ^^^^ expected `usize`, found `u32`
     |
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
     |
1091 |     debug_assert!(n < bits.try_into().unwrap() && n != 0);
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1098:25
     |
1098 |     let lshift = bits - n;
     |                         ^ expected `u32`, found `usize`

error[E0277]: cannot subtract `usize` from `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1098:23
     |
1098 |     let lshift = bits - n;
     |                       ^ no implementation for `u32 - usize`
     |
     = help: the trait `Sub<usize>` is not implemented for `u32`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1140:19
     |
1140 |     let rem = n % bits;
     |                   ^^^^ expected `usize`, found `u32`

error[E0277]: cannot mod `usize` by `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1140:17
     |
1140 |     let rem = n % bits;
     |                 ^ no implementation for `usize % u32`
     |
     = help: the trait `Rem<u32>` is not implemented for `usize`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1141:19
     |
1141 |     let div = n / bits;
     |                   ^^^^ expected `usize`, found `u32`

error[E0277]: cannot divide `usize` by `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1141:17
     |
1141 |     let div = n / bits;
     |                 ^ no implementation for `usize / u32`
     |
     = help: the trait `Div<u32>` is not implemented for `usize`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1197:23
     |
1197 |     debug_assert!(n < bits);
     |                       ^^^^ expected `usize`, found `u32`
     |
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
     |
1197 |     debug_assert!(n < bits.try_into().unwrap());
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1207:25
     |
1207 |     let rshift = bits - n;
     |                         ^ expected `u32`, found `usize`

error[E0277]: cannot subtract `usize` from `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1207:23
     |
1207 |     let rshift = bits - n;
     |                       ^ no implementation for `u32 - usize`
     |
     = help: the trait `Sub<usize>` is not implemented for `u32`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1259:19
     |
1259 |     let rem = n % bits;
     |                   ^^^^ expected `usize`, found `u32`

error[E0277]: cannot mod `usize` by `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1259:17
     |
1259 |     let rem = n % bits;
     |                 ^ no implementation for `usize % u32`
     |
     = help: the trait `Rem<u32>` is not implemented for `usize`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1260:19
     |
1260 |     let div = n / bits;
     |                   ^^^^ expected `usize`, found `u32`

error[E0277]: cannot divide `usize` by `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:1260:17
     |
1260 |     let div = n / bits;
     |                 ^ no implementation for `usize / u32`
     |
     = help: the trait `Div<u32>` is not implemented for `usize`

error[E0308]: mismatched types
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:2071:27
     |
2071 |     let rs = Limb::BITS - s;
     |                           ^ expected `u32`, found `usize`

error[E0277]: cannot subtract `usize` from `u32`
    --> /build/pastel-0.8.0-vendor.tar.gz/lexical-core/src/atof/algorithm/math.rs:2071:25
     |
2071 |     let rs = Limb::BITS - s;
     |                         ^ no implementation for `u32 - usize`
     |
     = help: the trait `Sub<usize>` is not implemented for `u32`

error: aborting due to 27 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `lexical-core`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
error: builder for '/nix/store/gcna0k7812a7f8i5vgdvx8wb0jpr7wnc-pastel-0.8.0.drv' failed with exit code 101;
       last 10 log lines:
       >
       > error: aborting due to 27 previous errors
       >
       > Some errors have detailed explanations: E0277, E0308.
       > For more information about an error, try `rustc --explain E0277`.
       > error: could not compile `lexical-core`
       >
       > To learn more, run the command again with --verbose.
       > warning: build failed, waiting for other jobs to finish...
       > error: build failed
       For full logs, run 'nix log /nix/store/gcna0k7812a7f8i5vgdvx8wb0jpr7wnc-pastel-0.8.0.drv'.
```

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build -A pastel`

**Expected behavior**

**Screenshots**

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@davidtwco

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.14-zen1, NixOS, 21.11.20210706.00c86ad (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210601_5985b8b`
 - channels(root): `"nixos-21.11pre299825.d8079260a30"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/misc/pastel/default.nix
+++ b/pkgs/applications/misc/pastel/default.nix
@@ -2,16 +2,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "pastel";
-  version = "0.8.0";
+  version = "0.8.1";
 
   src = fetchFromGitHub {
     owner = "sharkdp";
     repo = pname;
     rev = "v${version}";
-    sha256 = "00xxrssa3gbr5w2jsqlf632jlzc0lc2rpybnbv618ndy5lxidnw0";
+    sha256 = "12n1a9j61r4spx0zi2kk85nslv11j1s510asxqvj92ggqhr2s3sq";
   };
 
-  cargoSha256 = "0kkhj58q1lgsyj7hpy3sxg1jva9q51m0i7j60zfzhnjnirwcd0h8";
+  cargoSha256 = "12zachbg78ajx1n1mqp53rd00dzcss5cqhsq0119lalzc8b5zkrn";
 
   buildInputs = lib.optional stdenv.isDarwin Security;
 

